lib/heap_sort.rb

```diff
@@ -1,8 +1,105 @@
 
 
+# # This method uses a heap to sort an array.
+#  # Time Complexity: O(n log n) to add elements to a heap and swap them in order at each level
+#  # Space Complexity: O(log n) for recursive heap_up in .add
+def heapsort(list)
```

Comment:
👍 , except that the space complexity is O(n) since you're adding all the elements to a heap here.